src/trackinfoobject.cpp

```diff
@@ -37,12 +9,34 @@
 #include "track/keyutils.h"
 #include "util/compatibility.h"
 #include "util/cmdlineargs.h"
-#include "util/time.h"
 #include "util/math.h"
 #include "waveform/waveform.h"
 #include "library/coverartutils.h"
 #include "util/assert.h"
 
+#include <QDomNode>
+#include <QDomDocument>
+#include <QDomElement>
+#include <QFileInfo>
+#include <QDirIterator>
+#include <QFile>
+#include <QMutexLocker>
+#include <QString>
+#include <QtDebug>
+#include <QRegExp>
```

Comment:
Thanks for reviewing, Daniel :)

From my own experience I strongly recommend to include from most specific to least specific, i.e. to place project specific includes before any system includes. Reason: This order will reveal any hidden dependencies in header files that are not self-contained.

Unfortunately this is the reverse order that both the Mixxx and Google C++ coding style guides recommend! I really don't understand why the "Google C++ Style Guide" demands "self-contained" headers, but on the other hand proposes the wrong include order from least specific to most specific that prevents the compiler from detecting any violations of the self-containment rule.

Any ideas?